File: scripts/R/ch10_descriptive_stats.R
library(readr)
library(dplyr)

if (!file.exists("output/clean_clinic.csv")) {
  source("scripts/R/ch09_clean_recode.R")
}
clean_clinic <- read_csv("output/clean_clinic.csv", show_col_types = FALSE)

summary <- clean_clinic %>%
  group_by(group_clean) %>%
  summarise(
    n = n(),
    missing_age = sum(is.na(age)),
    mean_age = mean(age, na.rm = TRUE),
    median_age = median(age, na.rm = TRUE),
    sd_age = sd(age, na.rm = TRUE),
    mean_change = mean(change, na.rm = TRUE),
    median_change = median(change, na.rm = TRUE),
    .groups = "drop"
  )
print(summary)
write_csv(summary, "output/table1_group_summary.csv")

File: scripts/python/ch10_descriptive_stats.py
import pandas as pd

try:
    clean_clinic = pd.read_csv("output/clean_clinic.csv")
except FileNotFoundError:
    import subprocess, sys
    subprocess.run([sys.executable, "scripts/python/ch09_clean_recode.py"], check=True)
    clean_clinic = pd.read_csv("output/clean_clinic.csv")

summary = clean_clinic.groupby("group_clean").agg(
    n=("id", "size"),
    missing_age=("age", lambda x: x.isna().sum()),
    mean_age=("age", "mean"),
    median_age=("age", "median"),
    sd_age=("age", "std"),
    mean_change=("change", "mean"),
    median_change=("change", "median"),
).reset_index()
print(summary)
summary.to_csv("output/table1_group_summary.csv", index=False)
